src/cascadia/TerminalApp/App.cpp

```diff
@@ -159,13 +162,76 @@ namespace winrt::TerminalApp::implementation
         {
             IInspectable g = res.Lookup(key);
             winrt::Windows::UI::Xaml::Style style = g.try_as<winrt::Windows::UI::Xaml::Style>();
+            _root.Style(style);
             _tabRow.Style(style);
         }
 
         // Apply the UI theme from our settings to our UI elements
         _ApplyTheme(_settings->GlobalSettings().GetRequestedTheme());
 
         _OpenNewTab(std::nullopt);
+
+        _root.Loaded({ this, &App::_OnLoaded });
+    }
+
+    // Method Description:
+    // - Show a ContentDialog with a single "Ok" button to dismiss. Looks up the
+    //   the title and text from our Resources using the provided keys.
+    // - Only one dialog can be visible at a time. If another dialog is visible
+    //   when this is called, nothing happens.
+    // Arguments:
+    // - titleKey: The key to use to lookup the title text from our resources.
+    // - textKey: The key to use to lookup the content text from our resources.
+    fire_and_forget App::_ShowOkDialog(const winrt::hstring& titleKey,
+                                       const winrt::hstring& textKey)
+    {
+        // DON'T release this lock in a wil::scope_exit. The scope_exit will get
+        // called when we await, which is not what we want.
+        auto gotLock = _dialogLock.try_lock();

```

Comment:
Now that C++17 is in use, and assuming I haven't misread how the lock works here, you could use an init-statement to combine the initialization and condition evaluation if you prefer:

`if (auto gotLock = _dialogLock.try_lock(); !gotLock)`

Reply:
(I was also curious why the code didn't just do "if (! _dialogLock.try_lock())", but I assume it was because it was desirable to have the result be in a local for debugging?)

Reply:
We try to avoid assigning variables in if statements.

Reply:
(qq: is there a reason?)

Reply:
> (qq: is there a reason?)

I have a vague memory of a previous Windows coding guideline telling me to not do that or something of that ilk. I can't find proof of it. Maybe time to revisit? I defer to @adiviness though on matters of optimal code practice.

Reply:
I thought this was just an _always_ coding guideline. Something about the fact that `if (foo = a)` and `if (foo == a)` are far too visually similar, so separating assignment from the conditional was preferred. (That might not be the reason, but it's _a_ reason). If that's guidance has changed I'd certainly be shocked

Reply:
Since this is a new c++17 feature I'm not seeing anything specifically about it in the windows coding guidelines. I can ask, I'm curious now.

Reply:
I believe it to be a valuable convention, especially around the use of scoped resources (locked weak pointers, lock guards, etc.). I'd also be interested in seeing what isocpp says about it?

Reply:
I of course complain about this, then go write a bunch of code that does
```c++
    if (auto name{ json[NAME_KEY.data()] })
    {
        result._schemeName = name.asString();
    }
```

So whatever way this discussion goes will be very insightful

Reply:
The people in charge of the Windows coding guidelines seem to like it so I guess we should start using it where applicable.